### DISCUSSION TITLE
HFS+ Support?

### DISCUSSION THREAD

**Discussion by Stefankante** [outdoorbits/little-backup-box#198] [Q&A]
Just ask if it's possible to use HFS+ on Pi4 LBB to backup footage, BTW I want to ask first if HFS+ has reduced performance like NTFS.

**Comment by outdoorbits** [comment]
This is not supported yet. I never tried but hfs+ should need additional packages and a special command to mount it.
Do you want to use hfs+ as target? Is there any advantage compared to the available filesystems?

**Reply by Stefankante** [reply]
I find exFat harder to recover the footage if something happen to go bad especially with Exfat.
EXT4 is great in terms of speed and recover ability but I need to install paragon EXTfs on Macs also people who dealing with it footage will be so confuse.
NTFS - no no no with that speed.
So I curious about HFS+ which works great with production people but I need to test the speed.

**Reply by Stefankante** [reply]
HFS+ and NTFS saves countless time when data corrupted. It can be recover easily with disk drill in my experience.

**Comment by outdoorbits** [comment]
I just checked it. I will include hfs(+), hopefully next week.

**Comment by outdoorbits** [comment]
Thanks to the rainy weather, I was already able to upload the HFS/HFS Plus support. Please test it. Feedback is very welcome.

**Comment by Stefankante** [comment]
OH MY GODDDD!!!!! This is excellent!!!!

It transfer at the rate near or equal to Exfat or EXT4. Here's proof BTW it could go upward to 100+MB in great mood.
![IMG_3674](https://user-images.githubusercontent.com/35141284/229380961-d08261db-40bb-4012-b9b3-7dbb5183351c.jpg)

**Reply by Stefankante** [reply]
There's one problem, I can't format Mac Extended (Journaled) on my Mac as the LBB cannot write onto it and attempt with failed.
HFS+ is the same thing why I can't use it though. it also show HFSplus on display. Weird.

But I could use LBB to prep the drive anyway.

**Reply by Stefankante** [reply]
Seems like it's not the same but Paragon has option to format it, I will take note on that.

![Screen Shot 2566-04-03 at 05 19 32](https://user-images.githubusercontent.com/35141284/229382109-d9cd94e6-c441-4dc2-8cb0-b6786f3a1ee3.png)

**Comment by outdoorbits** [comment]
As much as I know both, APFS and Mac OS Extended, are not fully 
supported by linux. For APFS I found a project having read-only-access 
only. But have you seen, in Tools you can format HFS and HFS+.

**Comment by outdoorbits** [comment]
Is Mac OS Extended and HFS+ the same??? What if you format HFS+ by lbb, does your Mac recognize this as HFS+?

**Reply by Stefankante** [reply]
Apparently, HFS+ and Mac OS Extended are the same.
But Mac OS Extended has option to go with "Journaled" which Linux can read but can't write.
The HFS+ from LBB recognize as "Mac OS Extended" While normal Disk utility in Mac only has "Mac OS Extended (Journaled)" option to choose to format.
Also it has to be GUID not MBR as I tested yesterday, otherwise it won't write either.

My workaround is that who responsible to format that drive has to format in Macdrive (Windows) or Paragon NTFS (Mac) to get Mac OS Extended option. If not willing to use Web based Menu on LBB.

**Reply by Stefankante** [reply]
I wondered if this is limitation of Rpi bandwidth that could only deal with speed with 2 USB drives only cap around 100MB/s? BC I tried Offload file from 2 super fast drives and still cap at that speed, At 2.1GHz overclocked.

![339737342_641892751090758_559240506113153224_n](https://user-images.githubusercontent.com/35141284/229469553-4e024986-1210-4a78-8027-6d6fb4af5291.jpg)

**Comment by outdoorbits** [comment]
I never saw higher rates. It's far away from what to expect from USB 3. Maybe there are also some costs using rsync?

**Reply by Stefankante** [reply]
Maybe as I found on some discussions over internet.
[https://superuser.com/questions/109780/how-to-speed-up-rsync-between-two-local-disks](url)
Curious to see if it improve or if cp faster, BTW I'm so noob in coding. 😂

**Reply by Stefankante** [reply]
Both drive capable of read / write at Max each about ~260MB/s I have to try copy speed between drive.
<img width="822" alt="Screen Shot 2566-04-03 at 18 34 28" src="https://user-images.githubusercontent.com/35141284/229498580-5c49ea8a-abba-4263-99e5-87ce8fe0986f.png">

**Reply by Stefankante** [reply]
Maybe it capped around ~250MB/s for full bandwidth of Pi4 could handle via USB SSD 
divided by 2 with file system conversion then 120MB/s transfer between disks quite possibly explain the situation.

**Reply by Stefankante** [reply]
<img width="752" alt="Screen Shot 2566-04-03 at 20 07 09" src="https://user-images.githubusercontent.com/35141284/229519066-18b21cac-04eb-4a3e-a130-23a4c19e783b.png">

**Comment by outdoorbits** [comment]
This looks very interesting! Later I will implement this. I will tell 
you here as soon as the update is available.

**Comment by outdoorbits** [comment]
For local operations, rsync got --inplace, for cloud operation -z (compress). Please try it, I couldn't measure the expected significant speed up. Do you?

**Reply by Stefankante** [reply]
Based on Transfer between drives speed with dd and hdparm seems to capped around ~110MB/s max bc of hardware limitation.
It's acceptable though as I could transfer 81GB to backup hdd in 16 mins or less with such small setup. LBB seems to hover around that one (ExFat/EXT4/HFS+) 3.5inches external and SSD around 80-110MB/s.
